src/components/SearchBar/SearchBar.vue

```diff
@@ -135,6 +139,12 @@ export default {
   },
   mounted () {
     window.addEventListener('click', this.onClickOutside);
+
+    this.$refs?.searchContainer.addEventListener('focus', () => {
+      this.ringState = '4';
+      this.$refs.textInput.focus();
+    });
+    this.$refs?.searchContainer.addEventListener('click', () => this.ringState = 'none');
```

Comment:
I wonder if you really need to add event listeners to do this or if you can just use the regular `v-on` directives for the events (i.e. `@focus="handleFocus"` and `@click="handleClick"`) on the `searchContainer`.

Reply:
ah. yes, probably; it took us some time to land on this solution so we didn't think of this 😓 Pat let's try it!

Reply:
I added the directives and it works perfectly! So much cleaner!